tests/Mixins/DownloadCollectionTest.php

```diff
@@ -26,4 +27,26 @@ public function can_store_a_collection_as_excel()
             $response->headers->get('Content-Disposition')
         );
     }
+
+    /**
+     * @test
+     */
+    public function can_store_a_collection_with_headers_as_excel()
+    {
+        $collection = new Collection([
+            ['column_1' => 'test', 'column_2' => 'test'],
+            ['column_1' => 'test', 'column_2' => 'test'],
+        ]);
+
+        $response = $collection->downloadExcel('collection-headers-download.xlsx', Excel::XLSX, true);
+
+        $array = $this->readAsArray($response->getFile()->getPathName(), Excel::XLSX);
+
+        $this->assertEquals($collection->collapse()->keys()->all(), collect($array)->first());
```

Comment:
I would prefer an explicit key check, instead of using the same method as in the implementation. Would make it more explicit to understand what is being tested.